### ISSUE TITLE
Add rule to warn on shortcut references without definitions

### ISSUE THREAD

**Issue by wooorm** [wooorm/remark-lint#35]
```md
<!-- Good -->
[foo][]

[foo]: https://example.com

<!-- Bad -->
[bar][]
```

(I think this was once in here, not sure where it went.)
